src/zap_hooks/helpers/auth.py

```diff
@@ -114,9 +114,9 @@ def validate_authentication_url(driver, url):
     log(f"Validating authentication url: {url}")
     url_found = False
     for request in driver.requests:
-        if request.response and url in request.url:
+        if request.response and (url in request.url or search(url, request.url) is not None):
```

Comment:
This supports regular URL (current functionality) or a regex expression.

Reply:
Will validate all requests that match regex, not just the first. It's a quick change if we'd prefer to only validate the first match.